mempool/mempool.go

```diff
@@ -69,11 +74,42 @@ func (t *memTxnList) pop() (BroadcastedTransaction, error) {
 	return headNode.Txn, nil
 }
 
+func (t *memTxnList) popBatch(numToPop int) ([]BroadcastedTransaction, error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	if t.head == nil {
+		return nil, ErrTxnPoolEmpty
+	}
+
+	// Limit numToPop to the actual number of transactions available
+	if numToPop > t.len {
+		numToPop = t.len
+	}
+
+	result := make([]BroadcastedTransaction, 0, numToPop)
+	current := t.head
+
+	for range numToPop {
+		result = append(result, current.Txn)
+		current = current.Next
+	}
```

Comment:
We have benchmarked results that this approach is several times slower (in the miliseconds range) than
```go
for i := range numToPop {
   result[i] = current.Txn
}
```